metricbeat/helper/prometheus/module.go

```diff
@@ -0,0 +1,144 @@
+package prometheus
+
+import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	defaultScheme = "http"
+	defaultPath   = "/metrics"
+)
+
+var (
+	// HostParser validates Prometheus URLs
+	HostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		DefaultPath:   defaultPath,
+	}.Build()
+
+	expectedFlag = flag.Bool("update_expected", false, "Update prometheus expected files")
```

Comment:
Is this only used for testing? If yes, should it really be registered here?

Reply:
You are right, if it is defined here it will appear in the production binary. Testing code should be in a submodule to avoid that.

Reply:
As we use cobra, common flags are not really "leaked" to the binary, I'm ok with moving this though